libraries/Microsoft.Bot.Builder.AI.QnA/QnAMaker.cs

```diff
@@ -139,54 +108,101 @@ public async Task<QueryResult[]> GetAnswersAsync(ITurnContext turnContext, QnAMa
+                if (queryOptions.Top != hydratedOptions.Top && queryOptions.Top != 0)
+                {
+                    hydratedOptions.Top = queryOptions.Top;
+                }
 
-            var isLegacyProtocol = _endpoint.Host.EndsWith("v2.0") || _endpoint.Host.EndsWith("v3.0");
+                if (queryOptions.StrictFilters?.Length > 0)
+                {
+                   hydratedOptions.StrictFilters = queryOptions.StrictFilters;
+                }
+
+                if (queryOptions.MetadataBoost?.Length > 0)
+                {
+                   hydratedOptions.MetadataBoost = queryOptions.MetadataBoost;
+                }
+            }
+
+            return hydratedOptions;
+        }
 
-            if (isLegacyProtocol)
+        private void ValidateOptions(QnAMakerOptions options)
+        {
+            if (options.ScoreThreshold == 0)
             {
-                request.Headers.Add("Ocp-Apim-Subscription-Key", _endpoint.EndpointKey);
+                options.ScoreThreshold = 0.3F;
             }
-            else
+
+            if (options.Top == 0)
             {
-                request.Headers.Add("Authorization", $"EndpointKey {_endpoint.EndpointKey}");
+                options.Top = 1;
             }
 
-            AddUserAgent(request);
+            if (options.ScoreThreshold < 0 || options.ScoreThreshold > 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(options.ScoreThreshold), "Score threshold should be a value between 0 and 1");
+            }
 
-            var response = await _httpClient.SendAsync(request).ConfigureAwait(false);
-            if (!response.IsSuccessStatusCode)
+            if (options.Top < 1)
             {
-                return null;
+                throw new ArgumentOutOfRangeException(nameof(options.Top), "Top should be an integer greater than 0");
             }
 
-            var jsonResponse = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+            if (options.StrictFilters == null)
+            {
+                options.StrictFilters = new Metadata[] { };
+            }
 
-            var results = isLegacyProtocol ?
-                ConvertLegacyResults(JsonConvert.DeserializeObject<InternalQueryResults>(jsonResponse))
-                    :
-                JsonConvert.DeserializeObject<QueryResults>(jsonResponse);
+            if (options.MetadataBoost == null)
+            {
+                options.MetadataBoost = new Metadata[] { };
+            }
+        }
 
-            foreach (var answer in results.Answers)
+        private async Task<QueryResult[]> QueryQnaServiceAsync(Activity messageActivity, QnAMakerOptions options)
+        {
+            var request = BuildRequest(messageActivity, options);
+
+            var response = await _httpClient.SendAsync(request).ConfigureAwait(false);
+
+            if (!response.IsSuccessStatusCode)
```

Comment:
why not just use the EnsureSuccess and let the http library do the throw for you?
did you try that first?
I would try that and see that you get the message you'd like and then only if its broken do your own thing (I don't think its broken:-))

Reply:
`EnsureSuccessStatusCode` passes unit tests and I also ran qna sample, pointing to changed qna code. 

Throws errors as expected! woohoo :)